pkg/arvo/ted/ddiff.hoon

```diff
@@ -0,0 +1,304 @@
+  ::
+  ++  format-unique-children
+    |=  [bem=beam ark=arch children=(set @t)]
+    =/  m  (strand ,(list path))
+    ^-  form:m
+    =/  children=(list @t)  ~(tap in children)
+    =|  acc=(list path)
+    |-
+    =*  loop  $
+    ^-  form:m
+    ?~  children
+      (pure:m acc)
+    ::  the %t care gives all paths with the specified prefix
+    ;<  res=(list path)  bind:m  (get-from-clay bem(s (snoc s.bem i.children)) (list path) %t)
+    %=  loop
+      acc  (weld res acc)
+      children  t.children
+    ==
+  ::  +diff-file-contents: diff two files at specified beams,
+  ::  producing a vase of (unit diff-type)
+  ++  diff-file-contents
+    =<
+    |=  [a=beam a-arch=arch b=beam b-arch=arch]
+    =/  m  (strand ,(unit diff-type))
+    ^-  form:m
+    ?:  =(fil.a-arch fil.b-arch)
+      (pure:m *(unit diff-type))
+    ?~  fil.a-arch
+      :: only b has contents
+      %-  pure:m
+      ^-  (unit diff-type)
+      %-  some
+      :^    s.a
+          s.b
+        %txt-diff
+      :_  ~
+      ^-  (unce cord)
+      :+  %|
+        ~
+      ~[(format-file-content-missing s.b q.b)]
+    ?~  fil.b-arch
+      :: only a has contents
+      %-  pure:m
+      ^-  (unit diff-type)
+      %-  some
+      :^    s.a
+          s.b
+        %txt-diff
+      :_  ~
+      ^-  (unce cord)
+      :+  %|
+        ~[(format-file-content-missing s.a q.a)]
+      ~
+    ::  have two file contents - check that they have
+    ::  the same mark.
+    =/  mar=mark  -:(flop s.a)
+    ?:  !=(mar -:(flop s.b))
+      (strand-fail:strandio %files-not-same-type >s.a< >s.b< ~)
+    ;<  =a=cage     bind:m  (get-file a)
+    ;<  =b=cage     bind:m  (get-file b)
+    ;<  =dais:clay  bind:m  (build-mark:strandio -.a mar)
+    ::  for txt-diff we produce an actual diff with type (urge cord).
+    ::  for all other marks we just take the mug)
+    %-  pure:m
+    ?:  =(form:dais %txt-diff)
+      ^-  (unit diff-type)
+      %-  some
+      :^    s.a
+          s.b
+        %txt-diff
+      !<((urge cord) (~(diff dais q.a-cage) q.b-cage))
+    ^-  (unit diff-type)
+    %-  some
+    :^    s.a
+        s.b
+      %other
+    ::  For some reason, vases for identical files on different desks
+    ::  can sometimes have different types. for this reason, we only
+    ::  take the mug of the data.
+    [(mug q.q.a-cage) (mug q.q.b-cage)]
+    |%
+    ++  format-file-content-missing
+      |=  [p=path d=desk]
+      ^-  cord
+      %-  crip
+      ;:  weld
+        "only "
+        <p>
+        " in desk "
+        <d>
+        " has file contents"
+      ==
+    --
+  --
+::  +format-paths: helper to combine two paths into a tape
+::
+++  format-paths
+  |=  [a=path b=path]
+  ^-  tape
+  ;:(weld <a> " " <b>)
+::  +format-directory-diff: helper for producing a tank based on
+::  a %directory-diff
+::
+++  format-directory-diff
+  |=  [paths=(list path) =desk]
+  ^-  tang
+  =/  prefix=tape  ;:(weld "only in " (trip desk) ": ")
+  (turn paths |=(p=path >(weld prefix <p>)<))
```

Comment:
If you change this from circumfix `><` to prefix `leaf+`, it'll display each line without the double quotes, which will look nicer.

Reply:
`+smyt` is `$-(path tank)`, so I'd recommend something like:

```
%+  turn  paths
|=  p=path  ^-  tank
[%rose  [": " ~ ~] leaf+"only in %kids" (smyt p)]
```